x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java

```diff
@@ -459,12 +470,16 @@ public void onResponse(PersistentTasksCustomMetadata.PersistentTask<PersistentTa
                     if (countDown.decrementAndGet() == 0) {
                         logger.info("All downsampling tasks completed [" + numberOfShards + "]");
                         updateTargetIndexSettingStep(request, listener, sourceIndexMetadata, downsampleIndexName, parentTask);
+                        downsampleMetrics.recordLatencyMaster(getDurationInMillis(), DownsampleMetrics.ActionStatus.SUCCESS);
```

Comment:
Do we want to take into account the time it takes to perform the refresh and force merge? If so then we should move this to `ForceMergeActionListener`.

Reply:
Moved, and added also in the failure handlers, ptal.